### PR TITLE
Draggable Table columns

### PR DESCRIPTION
**Pull request by FritzHoing**
## Description

This MR introduces a new property that can make the table columns draggable.

draggable-column

## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the
  [BSD-2-Clause](https://github.com/terrestris/react-geo/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/react-geo/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [Code of Conduct](https://github.com/terrestris/react-geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally).
- [x] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

File: src/Grid/FeatureGrid/FeatureGrid.tsx [hunk 1]
```diff
@@ -41,11 +62,48 @@ const rowKeyClassNamePrefix = 'row-key-';
...
 const cellRowHoverClassName = 'ant-table-cell-row-hover';
 
+const DragIndexContext = createContext<DragIndexState>({ active: -1, over: -1 });
+
+const dragActiveStyle = (dragState: DragIndexState, id: string) => {
+  const { active, over, direction } = dragState;
+  // drag active style
+  let style: React.CSSProperties = {};
+  if (active && active === id) {
+    style = { backgroundColor: 'gray', opacity: 0.5 };
```
Comment (dnlkoch) [inline src/Grid/FeatureGrid/FeatureGrid.tsx:72]: Not sure if possible, but there are several inline css variations. Can we move them to the appropriate `*.less` files?

File: src/Grid/FeatureGrid/FeatureGrid.tsx [hunk 2]
```diff
@@ -62,9 +120,22 @@ export const FeatureGrid = <T extends AnyObject = AnyObject,>({
...
   const [selectedRowKeys, setSelectedRowKeys] = useState<Key[]>([]);
+  const [dragIndex, setDragIndex] = useState<DragIndexState>({ active: -1, over: -1 });
+  const [featureColumns, setFeatureColumns] = useState<ColumnType<T>[]>(initialColumns);
+
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 1,
+      },
+    }),
```
Comment (dnlkoch) [inline src/Grid/FeatureGrid/FeatureGrid.tsx:137]:
```suggestion
    useSensor(PointerSensor, {
      activationConstraint: {
        distance: 1
      }
    })
```

File: src/Grid/FeatureGrid/FeatureGrid.tsx [hunk 3]
```diff
@@ -241,17 +312,27 @@ export const FeatureGrid = <T extends AnyObject = AnyObject,>({
...
     return columnDefs;
-  };
+  }, [columns, attributeBlacklist, features]);
+
+  useEffect(() => {
+    const columnDefs = getColumnDefs().map((column, i) => ({
+      ...column,
+      key: `${i}`,
+      onHeaderCell: () => ({ id: `${i}` }),
+      onCell: () => ({ id: `${i}` }),
```
Comment (dnlkoch) [inline src/Grid/FeatureGrid/FeatureGrid.tsx:322]:
```suggestion
      onCell: () => ({ id: `${i}` })
```

File: src/Grid/FeatureGrid/FeatureGrid.tsx [hunk 4]
```diff
@@ -412,23 +497,88 @@ export const FeatureGrid = <T extends AnyObject = AnyObject,>({
...
+        return { id: key };
+      } else {
+        return key;
+      }
+    });
+  };
+
+  const draggableComponents = {
+    header: { cell: TableHeaderCell },
+    body: { cell: TableBodyCell },
```
Comment (dnlkoch) [inline src/Grid/FeatureGrid/FeatureGrid.tsx:537]:
```suggestion
    body: { cell: TableBodyCell }
```

File: src/Grid/FeatureGrid/FeatureGrid.tsx [hunk 5]
```diff
@@ -412,23 +497,88 @@ export const FeatureGrid = <T extends AnyObject = AnyObject,>({
...
+    setDragIndex({ active: -1, over: -1 });
+  };
+
+  const onDragOver = ({ active, over }: DragOverEvent) => {
+    const activeIndex = featureColumns.findIndex((i) => i.key === active.id);
+    const overIndex = featureColumns.findIndex((i) => i.key === over?.id);
+    setDragIndex({
+      active: active.id,
+      over: over?.id,
+      direction: overIndex > activeIndex ? 'right' : 'left',
```
Comment (dnlkoch) [inline src/Grid/FeatureGrid/FeatureGrid.tsx:517]:
```suggestion
      direction: overIndex > activeIndex ? 'right' : 'left'
```

File: src/Grid/FeatureGrid/FeatureGrid.tsx [hunk 6]
```diff
@@ -26,7 +32,22 @@ import {
...
+}
+
+interface DragIndexState {
+  active: UniqueIdentifier;
+  over: UniqueIdentifier | undefined;
+  direction?: 'left' | 'right';
+}
+
 type OwnProps = {
+  draggableColumns?: boolean;
```
Comment (dnlkoch) [inline src/Grid/FeatureGrid/FeatureGrid.tsx:50]: Can we provide a short description for the api docs for the newly added property?

File: src/Grid/FeatureGrid/FeatureGrid.tsx (hunk 1 above)
Comment (FritzHoing) [reply src/Grid/FeatureGrid/FeatureGrid.tsx:75]: I think that might be possible but the component would need some refactoring. I could address that, aswell as the <th></th> return in a follow-up.

**Review by hwbllmnn — APPROVED** [review]

**Pull request merged**